### ISSUE TITLE
Error: The LHS of `:=` must be a string or a symbol

### ISSUE THREAD

**Issue by wolski** [tidyverse/dplyr#3706]
I am running the following code which is supposed to create a column name produced by summarize using paste. 

```
> data <- data.frame(A=c("A","B","C"), B=c(1,2,3))
> data %>% group_by(A) %>% summarize(paste("hello","world",sep="_"):= n())
Error: The LHS of `:=` must be a string or a symbol
```
Although paste evaluates to "character".  The code below works.

```
data <- data.frame(A=c("A","B","C"), B=c(1,2,3))
dD<- paste("hello","world",sep="_")
data %>% group_by(A) %>% summarize(dD:= n())
```

I am guessing this is related to NSE since not a bug but is there a way to make it work?

**Comment by cderv** [comment]
Yes it is related to NSE. You in the first case, you need to ask to unquote using `!!` your `paste` call to make it works.
``` r
library(dplyr, , warn.conflicts = FALSE)
data <- data.frame(A=c("A","B","C"), B=c(1,2,3))
data %>% group_by(A) %>% summarize( !! paste("hello", "world", sep = "_") := n() )
#> # A tibble: 3 x 2
#>   A     hello_world
#>   <fct>       <int>
#> 1 A               1
#> 2 B               1
#> 3 C               1
```

Created on 2018-07-05 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

In the second case, you evaluated `paste("hello","world",sep="_")` to assign it in `dD` so that LHS is really a string. In the first case LHS is not a string, it is a call (or something else but not a string)

Hopes it helps and it is clear. 

For this type of question, do not hesitate to post on http://community.rstudio.com/. There are a large community who will be able to explain things like this. Github issues are better suited for bugs or feature request. Thanks!
